docs: Deprecations for connect-native SDK and specific connect native APIs (#17937)
hashicorp/consul: website/content/docs/release-notes/consul/v1_16_x.mdx

  Consul's API gateway is the recommended alternative to ingress gateway. For ingress gateway features not currently supported by API gateway, equivalent functionality will be added to API gateway over the next several releases of Consul.

> - **Connect Native Golang SDK:** The Connect Native [Golang SDK](https://github.com/hashicorp/consul/tree/main/connect) is deprecated and will be removed in a future release. No further enhancements or maintenance is expected in the future releases. The eventual removal of the SDK is dependent on the delivery of proxyless GRPC via XDS which is tracked on [GH-10339](https://github.com/hashicorp/consul/issues/10339). 

Reviewer: ```suggestion
- **Connect Native Golang SDK:** The Connect Native [Golang SDK](https://github.com/hashicorp/consul/tree/main/connect) is deprecated and will be removed in a future release. No further enhancements or maintenance is expected in the future releases. We will remove the SDK when the long term replacement, proxy-less gRPC connections to Consul xDS, is delivered. Refer to [GH-10339](https://github.com/hashicorp/consul/issues/10339) for additional information and to track progress toward the replacement functionality. 
```

- **Connect Native APIs:** The following APIs for Connect Native are deprecated: 
  - `v1/agent/connect/authorize` - used by the SDK to perform intention based authorization checks
  - `v1/agent/connect/ca/leaf` - used by the SDK to get a leaf cert for a locally registered service
  - `v1/agent/connect/ca/roots` - use to retrieved cached CA roots form the local client agent
  

>   Currently both `v1/agent/connect/authorize` and `v1/agent/connect/ca/leaf` have corresponding gRPC APIs. Removal of these APIs will be dependent on delivering the gRPC API for `v1/agent/connect/ca/roots` and HTTP endpoints for all three APIs for easier consumption by users. 

Reviewer: ```suggestion
  The `v1/agent/connect/authorize` and `v1/agent/connect/ca/leaf` endpoints have corresponding gRPC APIs. We will remove these APIs when the gRPC API for `v1/agent/connect/ca/roots` and HTTP endpoints for all three APIs are usable. 
```
